blocklylib-core/src/main/java/com/google/blockly/model/FieldNumber.java

```diff
@@ -182,14 +190,42 @@ public void setConstraints(double min, double max, double precision) {
                 char[] sigDigitsFormat = new char[significantDigits];
                 Arrays.fill(sigDigitsFormat, '#');
                 sb.append(sigDigitsFormat);
-                mFormatter = new DecimalFormat(sb.toString());
+                mFormatter = new DecimalFormat(sb.toString(), PERIOD_AS_DECIMAL);
             }
         }
 
         setValueImpl(mValue, true);
     }
 
     /**
+     * Retrieves (possibly constructing) a NumberFormat configured for both the field constraints
+     * and the provided Locale.
+     * @param locale The locale to construct a number formatter for.
+     * @return A NumberFormat configured for both the field constraints and the Locale.
+     */
+    public NumberFormat getNumberFormatForLocale(Locale locale) {
+        if (!hasPrecision()) {
+            return NumberFormat.getInstance(locale);
+        }
+        if (mIntegerPrecision) {
+            return NumberFormat.getIntegerInstance(locale);
+        }
+
+        String precisionStr = NAIVE_DECIMAL_FORMAT.format(mPrecision);
+        int decimalChar = precisionStr.indexOf('.');
+        if (decimalChar == -1) {
+            return NumberFormat.getIntegerInstance(locale);
+        }
+
+        int significantDigits = precisionStr.length() - decimalChar;
+        StringBuilder sb = new StringBuilder("0.");
```

Comment:
instead of building a format string can you just call setMaximumFractionDigits()? https://developer.android.com/reference/java/text/DecimalFormat.html#setMaximumFractionDigits(int)